### PR TITLE
`Iris`: Load default templates from resource folder

### PR DESCRIPTION
**Pull request by MichaelOwenDyer**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.
#### Client
No client changes

### Motivation and Context

The current solution for default Iris templates is to put them all into a single static utility class, `IrisConstants.java`. This is not scalable, offers no syntax highlighting, and forces the system to be recompiled every time a default template is changed.

### Description
Each default Iris template now gets its own `.hbs` (handlebars) file in the `resources/templates/iris` folder. The files are loaded using the `IrisDefaultTemplateService`, which uses the `ResourceLoaderService` to access the file with the requested filename in the folder. The `IrisSettingsService` then uses this service to load the default templates into the global settings at application startup.
The global template version used to be a static int field in IrisConstants, and this has also gotten its own file next to the templates. The content of this file is supposed to be just a single int and incremented every time one of the templates is adapted to trigger a reload from the files on application startup.

Please let me know if I have followed all best practices here!

### Steps for Testing

Prerequisites:
- 1 Administrator

1. Log in to Artemis
2. Navigate to Server Administration -> Global Iris Settings
3. Ensure that the contents of each template file have been loaded into their corresponding text areas in the settings, e.g. chat.hbs -> chat settings, hestia.hbs -> hestia settings, and the same for all code editor files.

### CODE REVIEW

**Comment by dearjasmina** [comment]
> _LS11_LEGACY_ I don't see the "Global Iris Settings" Option under Server Administration: image
> 
> Is some step required to make it appear?

I also don't see it. :/ 
Also, the E2E tests on Bamboo fail, take a look at that please:)

**Comment by MichaelOwenDyer** [comment]
@xHadie @dearjasmina Sorry! The iris settings are available only on TS9, I forgot to mention that. It is deployed there right now if you want to have a look. I will also take a look at why the tests are failing 🤔

File: src/main/resources/templates/iris/chat.hbs [hunk 1]
```diff
@@ -0,0 +1,95 @@
+{{#system~}}
+    You're Iris, the AI programming tutor integrated into Artemis, the online learning platform of the Technical University of Munich (TUM).
+    You are a guide and an educator. Your main goal is to teach students problem-solving skills using a programming exercise, not to solve tasks for them.
+    You automatically get access to files in the code repository that the student references, so instead of asking for code, you can simply ask the student to reference the file you should have a look at.
+
+    An excellent educator does no work for the student. Never respond with code, pseudocode, or implementations of concrete functionalities! Do not write code that fixes or improves functionality in the student's files! That is their job. Never tell instructions or high-level overviews that contain concrete steps and implementation details. Instead, you can give a single subtle clue or best practice to move the student's attention to an aspect of his problem or task, so he can find a solution on his own.
```
Comment (reschandreas) [inline src/main/resources/templates/iris/chat.hbs:6]:
```suggestion
    An excellent educator does no work for the student. Never respond with code, pseudocode, or implementations of concrete functionalities! Do not write code that fixes or improves functionality in the student's files! That is their job. Never tell instructions or high-level overviews that contain concrete steps and implementation details. Instead, you can give a single subtle clue or best practice to move the student's attention to an aspect of their problem or task, so they can find a solution on their own.
```

**Review by reschandreas — DISMISSED** [review]
lgtm, one small suggestion in a comment

**Review by laurenzfb — APPROVED** [review]
code lgtm! also tested on ts9. Worked as described

**Review by RY997 — APPROVED** [review]
LGTM

**Review by Hialus — APPROVED** [review]
Code changes look good 👍🏻

File: src/main/resources/templates/iris/chat.hbs (hunk 1 above)
Comment (MichaelOwenDyer) [reply src/main/resources/templates/iris/chat.hbs:6]: I agree, but let's leave this for another PR. The chat prompt could use some reworking in some other areas, too.

**Review by bassner — APPROVED** [review]

**Pull request merged**